### ISSUE TITLE
Unclear exception handling

### ISSUE THREAD

**Issue by baaltos** [buttplugio/buttplug-js#257]
Hello,

I am currently trying to use this library within a node-red configuration node.
I did manage to open a connection using the ButtplugClient and even send commands, but I am running into issues when handling connection error.

Whenever I try to connect using `client.connect` and the Server is not reachable (intiface being down for example) the code throws an exception that I can't catch.
I tried the following setups:

```js
# Setup 1
this.client.connect(connector).catch(err => console.log(err))

# Setup 2
try {
  await this.client.connect(connector)
} catch (err) {
  console.log(err)
}

# Setup 3 
process.on('uncaughtException', (err) => {console.log(err)})
```

None of the setups results in any exception handling code being executed.

Am I doing something wrong?

**Comment by baaltos** [comment]
I replicated the third setup in a normale JS file and managed to catch the exception.

But it does feel wrong that the exception can't be caught by the `.catch()` function that is executed on the connect result (Setup 1)

**Comment by qdot** [comment]
I would not be surprised if this is me fucking up async try/catch versus promise then/catch again. I'll try to revisit this and maybe actually write a test this time.

**Comment by baaltos** [comment]
Hi @qdot, I did try my hand at a fix. It seems to work locally in the tests and if i import it in node-red.
See PR #258

**Comment by qdot** [comment]
Finally merged in v3.2.2, going out to npm now. Thanks!
